src/test/java/teammates/test/cases/util/SanitizationHelperTest.java

```diff
@@ -358,4 +358,24 @@ public void testDesanitizeIfSanitized() {
         assertEquals("should return same unsanitized string if given unsanitized string containing sanitized sequences",
                 unsanitized, SanitizationHelper.desanitizeIfHtmlSanitized(unsanitized));
     }
+
+    @Test
+    public void testSanitizeForLogMessage() {
+        assertNull("should return null if given null", SanitizationHelper.sanitizeForLogMessage(null));
+
+        String unsanitized = "<span class=\"text-danger\"> A <span class=\"bold\">typical</span> log  message <br>"
+                + " It contains some <script>dangerous</script> elements </span>";
+        String correctSanitized =
+                "<span class=\"text-danger\"> A <span class=\"bold\">typical</span> log  message <br>"
+                + " It contains some &lt;script&gt;dangerous&lt;&#x2f;script&gt; elements </span>";
+        assertEquals("Should escape HTML special characters"
+                + "other than in <span class=\"text-danger\">, <span class=\"bold\"> and <br>",
+                correctSanitized, SanitizationHelper.sanitizeForLogMessage(unsanitized));
+
+        unsanitized = "Hmm. <span class=\"text-info\"> How about this? </span> and <span> this</span>";
+        correctSanitized =
+                "Hmm. &lt;span class=&quot;text-info&quot;&gt; How about this? </span> and <span> this</span>";
```

Comment:
The html tag is not well-formed :(
Since no better solution for this.

Reply:
yeah, this is ugly but there's no simple solution for this.

Reply:
Can be fixed together with sub-issue in #6502 - `give a standard way to generate statusToAdmin`

Reply:
Ideally, the log message should just contain plaintext but have a predefined form based on the result type. Then we can sanitize the whole log message, and use the predefined form to parse and format the message according to what we want.